server/src/auth/roles.guard.ts

```diff
@@ -19,6 +19,11 @@ export class RolesGuard implements CanActivate {
       return true;
     }
     const { user } = context.switchToHttp().getRequest();
-    return requiredRoles.some((role) => user.role === role);
+
+    if (!requiredRoles.some((role) => user.role === role)) {
+      throw new ForbiddenException('Forbidden — user role does not have access to this resource');
```

Comment:
Response code would already have Forbidden meaning. You can exclude the same word from the response text.